hibernate-core/src/main/java/org/hibernate/tool/schema/internal/DdlTransactionIsolatorProvidedConnectionImpl.java

```diff
@@ -49,5 +55,17 @@ public void prepare() {
 
 	@Override
 	public void release() {
+		JdbcConnectionAccess connectionAccess = jdbcContext.getJdbcConnectionAccess();
+		if( !( connectionAccess instanceof JdbcConnectionAccessProvidedConnectionImpl ) ) {
+			throw new IllegalStateException( "DdlTransactionIsolatorProvidedConnectionImpl should always use a JdbcConnectionAccessProvidedConnectionImpl" );
+		}
+		JdbcConnectionAccessProvidedConnectionImpl connectionAccessProvidedConnection =
+			(JdbcConnectionAccessProvidedConnectionImpl) connectionAccess;
+		try {
```

Comment:
There should be a comment explaining why passing 'null' to `JdbcConnectionAccess#releaseConnection`is OK. 

Is it valid to pass `null` (rather than the `Connection`) because we know that that connection was provided?

Reply:
While passing the connection to the `releaseConnection` method might be suitable for the `JdbcConnectionAccessConnectionProviderImpl`, `JdbcEnvironmentInitiator` or `NonContextualJdbcConnectionAccess`, it has no meaning for `JdbcConnectionAccessProvidedConnectionImpl` because, in this case, the connection is wrapped and we don't have access to it upon releasing via the `DdlTransactionIsolatorProvidedConnectionImpl`. 

That's why I added the check above so to make sure we don't pass `null` to a connection access that's not `JdbcConnectionAccessProvidedConnectionImpl`.